app/fruits/page.tsx

```diff
@@ -10,34 +10,29 @@ export const metadata = {
 
 export default function FruitsPage() {
   // get the cookie from the server
-  const fruitsCookie = cookies().get('fruitsCookie');
+  const fruitComments = parseJson(getCookie('fruitComments'));
 
-  // create a default value if cooke doesn't exist
-  let fruitsCookieParsed: CookieValue = [];
+  const currentComments = Array.isArray(fruitComments) ? fruitComments : [];
 
-  if (fruitsCookie) {
-    fruitsCookieParsed = JSON.parse(fruitsCookie.value);
-  }
+  const fruitsWithComments = fruits.map((fruit) => {
+    const fruitWithComment = { ...fruit, comment: '' };
 
-  const fruitsWithStars = fruits.map((fruit) => {
-    const fruitWithStars = { ...fruit, stars: 0 };
-
-    // i read the cookie and find the fruit
-    const fruitInCookie = fruitsCookieParsed.find(
+    // read the cookie and find the fruitNote
+    const fruitInCookie = currentComments.find(
```

Comment:
`fruitInCookie` is a confusing name to me

not sure what the best alternative would be here - maybe the variable can be avoided altogether...

Reply:
`matchingFruitFromCookie` is what I came up with below:

https://github.com/upleveled/next-js-example-winter-2023-vienna-austria/pull/15/files#r1197031142

also not great, but it is a bit more expressive at least

Reply:
https://github.com/upleveled/next-js-example-winter-2023-vienna-austria/commit/ae48d6bc14c2ca62a89811a454ac80f704e4b3e2